src/main.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+"""Play rock paper scissors on Twitter"""
+
+import os
+import enum
+import logging
+
+import tweepy  # type: ignore
+
+def configure_logger(module_name: str) -> logging.Logger:
+    """Configure the logger"""
+    logger = logging.getLogger(module_name)
+    formatter = logging.Formatter(fmt="%(levelname)s: %(message)s")
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    logger.setLevel(logging.INFO)
+    return logger
+
+LOG = configure_logger(__name__)
+
+class Weapon(enum.IntEnum):
+    """Enum of weapons"""
+    ROCK = 0
+    PAPER = 1
+    SCISSORS = 2
+
+    @property
+    def beaten_by(self) -> "Weapon":
+        """Creates a new weapon that will beat itself"""
+        return self.__class__((self.value + 1) % 3)
+
+    @property
+    def beats(self) -> "Weapon":
+        """Creates a new weapon that it will beat"""
+        return self.__class__((self.value - 1) % 3)
+
+class RockPaperScissors:
+    """Wrapper for the Twitter interface"""
+
+    MAX_COUNT = 1
+
+    def __init__(self) -> None:
+        auth = tweepy.OAuthHandler(os.environ["API_KEY"],
+                                   os.environ["API_SECRET"])
+        auth.set_access_token(os.environ["ACCESS_TOKEN"],
+                              os.environ["ACCESS_TOKEN_SECRET"])
+        self.api = tweepy.API(auth, wait_on_rate_limit=True,
+                              wait_on_rate_limit_notify=True)
+        self.me = self.api.me()  # pylint: disable=invalid-name
+        self.timeline = self.api.user_timeline(count=self.MAX_COUNT)
+
+    def _filter(self, tweet: tweepy.models.Status) -> bool:
+        if tweet.in_reply_to_user_id != self.me.id:
+            return False
+        if tweet.id in set(tweet.in_reply_to_status_id
+                           for tweet in self.timeline):
+            return False
+        weapons_in_tweet = set(weapon for weapon in Weapon
+                               if weapon.name in tweet.text.upper())
+        return len(weapons_in_tweet) == 1
+
+    @staticmethod
+    def _compose(tweet: tweepy.models.Status) -> dict:
+        weapon = set(weapon for weapon in Weapon
+                     if weapon.name in tweet.text.upper()).pop()
+        text = f"@{tweet.author.screen_name} {weapon.beaten_by.name.title()}."
+        return {"status": text, "in_reply_to_status_id": tweet.id}
+
+    def mentions(self) -> list:
+        """Get mentions"""
+        since_id = self.timeline[0].id
+        tweets = self.api.mentions_timeline(since_id=since_id,
+                                            count=self.MAX_COUNT)
```

Comment:
I considered using the streaming API, but don't really enjoy the idea of keeping a socket open forever. It would also mean that in the event of downtime, we wouldn't catch up to old tweets we may have missed.

So `MAX_COUNT` in production will for sure be much bigger. However I'm not sure if you picked the number 20 at random or not. There's actually another potential issue here, which is that according to the documentation, `mentions_timeline` returns at most 20 results. I'm not sure if this is by default or not and is worth testing. I was going to leave it as it is and see how it goes, and if it is overly constrained we can swap this method out for `search` or whatever, which is a little uglier but will get the job done (at, I believe, a higher API rate limiting cost).

Reply:
In terms of how to deploy, I am not too sure actually. I was going to run it on a regular server initially, because it's not really suitable for Heroku due to high calling frequency (presumably every ten minutes or so). A Lambda might work well...